plugins/scripts/inject-config.js

```diff
@@ -1,200 +1,255 @@
+  ];
+
+  if (
+    !isEmpty(config.inAppConsentStrategy) &&
+    validInAppStrategies.indexOf(config.inAppConsentStrategy) < 0
+  ) {
+    console.error(
+      "Optimove: invalid inAppConsentStrategy given in optimove.json, valid options are: " +
+        validInAppStrategies.join(", ")
+    );
+    return false;
+  } else if (isEmpty(config.inAppConsentStrategy)) {
+    config.inAppConsentStrategy = "in-app-disabled";
+  }
+
+  return config;
 }
 
 function isEmpty(val) {
-    return !val || !val.length;
+  return !val || !val.length;
 }
 
 function isString(val) {
-    return typeof val === 'string';
+  return typeof val === "string";
 }
-function readOptimoveSettings(context) {
-    const configFile = path.join(context.opts.projectRoot, 'optimove.json');
-
-    if (!fs.existsSync(configFile)) {
-        console.warn(
-            'Optimove: config file not found, please ensure optimove.json is present in your project root'
-        );
-        return false;
-    }
-
-    const configJson = fs.readFileSync(configFile, { encoding: 'utf-8' });
-    let config;
-
-    try {
-        config = JSON.parse(configJson);
-    } catch (e) {
-        console.error('Optimove: optimove.json file not valid JSON');
-        return false;
-    }
-
-    if (!isValidConfig(config)) {
-        return false;
-    }
-
-    return config;
+
+function hasPlatform(context, platform) {
+  return context.opts.platforms.indexOf(platform) > -1;
+}
+
+function renderTemplate(name, values) {
+  const templatePath = path.join(__dirname, name);
+  const template = fs.readFileSync(templatePath, { encoding: "utf-8" });
+  return replaceFields(template, values);
+}
+
+function replaceFields(str, fields) {
+  return Object.keys(fields).reduce(
+    (str, field) => str.replace(`{{${field}}}`, fields[field]),
+    str
+  );
 }
 
+
+// ===================== ANDROID SPECIFIC ======================
+
 function prepareAndroid(context, optimoveConfig) {
+  const config = createJsonWithDefaultValues(
+    optimoveConfig.optimoveCredentials,
+    optimoveConfig.optimoveMobileCredentials,
+    optimoveConfig.inAppConsentStrategy,
+    optimoveConfig.enableDeferredDeepLinking
+  );
+
+  writeOptimoveXml(context, config);
+
+  if (config.ENABLE_DEFERRED_DEEP_LINKING){
+    createOptimoveMainActivity(context);
+  }
+
+  writeGoogleServicesJson(context);
+}
+
+function createJsonWithDefaultValues(optimoveCredentials, optimoveMobileCredentials , inAppConsentStrategy, enableDeferredDeepLinking) {
+  return {
+    OPTIMOVE_CREDENTIALS:
+      !isEmpty(optimoveCredentials) &&
+      isString(optimoveCredentials)
+        ? optimoveCredentials
+        : "",
+    OPTIMOVE_MOBILE_CREDENTIALS:
+      !isEmpty(optimoveMobileCredentials) &&
+      isString(optimoveMobileCredentials)
+        ? optimoveMobileCredentials
+        : "",
+    IN_APP_STRATEGY: inAppConsentStrategy,
+    ENABLE_DEFERRED_DEEP_LINKING:
+      enableDeferredDeepLinking === 'true' || enableDeferredDeepLinking === true
```

Comment:
why do we handle this as both string and boolean?

Reply:
just they could type any, but guess no reason. updated to check only for bool